packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/richtext/richtext.component.ts

```diff
@@ -1,16 +1,55 @@
-﻿import { Component, OnInit } from '@angular/core';
+﻿import { Component, OnInit, ElementRef, Renderer2 } from '@angular/core';
 import { Field } from '@sitecore-jss/sitecore-jss-angular';
 import { SxaComponent } from '../sxa.component';
 
 @Component({
   selector: 'app-richtext',
   templateUrl: './richtext.component.html',
+  styles: [
+    `
+      :host(.footer-richtext) {
+        width: 17%;
+        display: inline-block;
+      }
+
+      :host(.footer-richtext) ::ng-deep > * {
+        width: 100%;
+        height: 100%;
+        display: flex;
+        flex-direction: column;
+      }
+
+      :host(.footer-richtext) ::ng-deep > *:nth-child(2) {
+        order: -1;
+      }
+
+      @media (max-width: 992px) {
+        :host(.footer-richtext) {
+          width: 100%;
+        }
+      }
+    `,
+  ],
 })
 export class RichTextComponent extends SxaComponent implements OnInit {
   text?: Field<string>;
 
+  constructor(private el: ElementRef, private renderer: Renderer2) {
+    super();
+  }
+
   ngOnInit() {
     super.ngOnInit();
     this.text = this.rendering.fields?.Text as Field<string>;
+
+    let parent = this.el.nativeElement.parentElement;
+
+    while (parent) {
+      if (parent.id === 'footer') {
```

Comment:
We can't use such condition and rely on specific parent component
it's a "hack", what will happen if id of the parent element is changed but it's actually the same "footer" placeholder?